packages/nuxt/src/app/composables/cookie.ts

```diff
@@ -14,7 +14,7 @@ type _CookieOptions = Omit<CookieSerializeOptions & CookieParseOptions, 'decode'
 
 export interface CookieOptions<T = any> extends _CookieOptions {
   decode?(value: string): T
-  encode?(value: T): string
+  encode?(value: string): string
```

Comment:
I think we want the ability for users to pass an encoder that handles non-strings...

```suggestion
  encode?(value: T): string
```

Reply:
Sure, but to avoid duplicate JSON.stringify encode will only receive string now, I've moved JSON.stringify from default encode to main function.

This ability is still preset for user, but encode will only receive strings from now on

```typescript
const rawCookie = typeof cookie.value === "string" ? cookie.value : JSON.stringify(cookie.value)
channel?.postMessage(opts.encode(rawCookie))
```